source/mesh_deformation/external_tool_interface.cc

```diff
@@ -0,0 +1,533 @@
+/*
+  Copyright (C) 2014 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/mesh_deformation/external_tool_interface.h>
+#include <aspect/simulator_signals.h>
+
+/*
+
+TODO:
+- PointDataOut: move to deal.II?
+- use name "nodes" instead of "evaluation points"
```

Comment:
I think the latter name is more generic. These points don't *have* to be node points of a mesh.